src/py/mat3ra/made/tools/build/utils.py

```diff
@@ -78,10 +81,98 @@ def merge_materials(
     distance_tolerance: float = 0.01,
     merge_dangerously=False,
 ) -> Material:
+    """
+    Merge multiple materials into a single material.
+
+    If some of the atoms are considered too close within a tolerance, only the last atom is kept.
+
+    Args:
+        materials (List[Material]): List of materials to merge.
+        material_name (Optional[str]): Name of the merged material.
+        distance_tolerance (float): The tolerance to replace atoms that are considered too close with respect
+            to the coordinates in the last material in the list, in angstroms.
+        merge_dangerously (bool): If True, the lattices are merged "as is" with no sanity checks.
+    Returns:
+        Material: The merged material.
+    """
     merged_material = materials[0]
     for material in materials[1:]:
         merged_material = merge_two_materials(
             merged_material, material, material_name, distance_tolerance, merge_dangerously
         )
 
     return merged_material
+
+
+def double_and_filter_material(material: Material, start: List[float], end: List[float]) -> Material:
+    """
+    Double the material and filter it by a box defined by the start and end coordinates.
+    Args:
+        material (Material): The material to double and filter.
+        start (List[float]): The start coordinates of the box.
+        end (List[float]): The end coordinates of the box.
+    Returns:
+        Material: The filtered material.
+    """
+    material_doubled = create_supercell(material, scaling_factor=[2, 1, 1])
+    return filter_by_box(material_doubled, start, end)
+
+
+def expand_lattice_vectors(material: Material, gap: float, direction: int = 0) -> Material:
+    """
+    Expand the lattice vectors of the material in the specified direction by the given gap.
+
+    Args:
+        material (Material): The material whose lattice vectors are to be expanded.
+        gap (float): The gap by which to expand the lattice vector.
+        direction (int): The index of the lattice vector to expand (0, 1, or 2).
+    """
+    new_lattice_vectors = material.lattice.vector_arrays
+    new_lattice_vectors[direction][direction] += gap
+    material.set_new_lattice_vectors(
+        lattice_vector1=new_lattice_vectors[0],
+        lattice_vector2=new_lattice_vectors[1],
+        lattice_vector3=new_lattice_vectors[2],
+    )
+    return material
+
+
+def merge_two_materials_laterally(
```

Comment:
maybe `stack` instead of `merge` here? We are really stacking the materials and creating a new one as opposed to "morphing" the materials into the first lattice

Reply:
Or `stack_two_materials_xy`